subprojects/core/src/main/java/org/gradle/internal/service/scopes/DefaultGradleUserHomeScopeServiceRegistry.java

```diff
@@ -75,7 +78,7 @@ public ServiceRegistry getServicesFor(final File gradleUserHomeDir) {
                     Services otherServices = servicesForHomeDir.values().iterator().next();
                     if (otherServices.count == 0) {
                         // Other home dir cached and not in use, clean it up
-                        CompositeStoppable.stoppable(otherServices.registry).stop();
```

Comment:
Imho this is kind of dryness and leaking (obsolete) impl. detail. If it’s Composite or Foo, who cares?  
Code needs to work and pass. This is something to investigate if mandatory, not to bother everybody by hiding and taking away from the actual thing.  

Assuming we just want to stop the reg, that’s it.